src/ucp/core/ucp_request.inl

```diff
@@ -363,6 +363,56 @@ void ucp_request_recv_generic_dt_finish(ucp_request_t *req)
     }
 }
 
+static UCS_F_ALWAYS_INLINE void
+ucp_request_memh_init(ucp_request_t *req, ucp_ep_h ep, ucp_mem_h memh)
+{
+    ucp_md_map_t *md_map_p = &req->send.state.dt.dt.contig.md_map;
+    uct_mem_h *uct_memh = req->send.state.dt.dt.contig.memh;
+    ucp_ep_config_t *ep_conf = ucp_ep_config(ep);
+    int i;
+    ucp_md_index_t md_idx;
+
+    assert(memh != NULL);
+
+    *md_map_p = 0;
+    for (i = 0; i < ep_conf->key.num_lanes; i++) {
+        md_idx = ep_conf->md_index[i];
+        if (md_idx != UCP_NULL_RESOURCE) {
+            *md_map_p |= UCS_BIT(md_idx);
+        }
+    }
+
+    i = 0;
+    ucs_for_each_bit(md_idx, *md_map_p) {
+        assert(memh->md_map & UCS_BIT(md_idx));
```

Comment:
so, there is a requirement for memh to be symmetric for ep config. Shall we state that in the documentation (ucp.h)?

Reply:
what does symmetric mean? user provided ucp memh has uct memhs for all mds but we only save those for active mds for ep

Reply:
if I read this correctly you expect that memh provided by the user will always contains UCT memh for all ep lanes/mds.
Is this expectation based on the fact that memh can be obtained using `ucp_mem_map` only, which always registers memory on all available mds?

Reply:
yes. ucp_mem_map was the case.

Reply:
So, I'd document this expectation (maybe in the comment for memh parameter in ucp_request_params)
@yosefe, wdyt?

Reply:
done